src/common.cu

```diff
@@ -870,8 +937,50 @@ int main(int argc, char* argv[]) {
            (unsigned long long)maxBytes);
     return -1;
   }
+
+#ifdef USE_C10D_GLOO
+  {
+    // Parse c10d GLOO distributed framework environment variables.
+    char *str = getenv("MASTER_ADDR");
+    if (str) {
+      std::string master_addr = str;
+      use_c10d_gloo = true;
+
+      str = getenv("MASTER_PORT");
+      uint16_t master_port = str ? static_cast<uint16_t>(std::stoi(str)) : 29500;
+
+      str = getenv("RANK");
+      int rank = str? std::stoi(str) : 0;
+
+      str = getenv("WORLD_SIZE");
+      int world_size = str ? std::stoi(str) : 1;
+
+      auto options = c10d::ProcessGroupGloo::Options::create();
+      // Create Gloo device that binds to any interface.
+      ::gloo::transport::tcp::attr tcp_attr;
+      tcp_attr.iface = "eth0";
```

Comment:
Why is the interface name hardcoded?

Reply:
Good catch. I will fix it to be configurable by an env variable. Thanks.